test/battle/ability/armor_tail.c

```diff
@@ -0,0 +1,40 @@
+#include "global.h"
+#include "test/battle.h"
+
+
+ASSUMPTIONS
+{
+    ASSUME(gMovesInfo[MOVE_QUICK_ATTACK].priority > 0);
+}
+
+DOUBLE_BATTLE_TEST("Armor Tail protects user from priority moves")
```

Comment:
I think we need @AsparagusEduardo's opinion, but do we really need four files with the majority of the code copy-pasted? Can't we just have the tests in common parametrized and add references to "tests for this ability are in file X", as is already the case in other test files I believe?

Reply:
> Can't we just have the tests in common parametrized and add references to "tests for this ability are in file X", as is already the case in other test files I believe?

Alright.

Reply:
combined them